.github/workflows/yocto-build-deploy.yml

```
@@ -121,37 +116,47 @@ on:
+        default: "arn:aws:iam::567579488761:role/balena-os-deploy-s3-access-Role-8r0SXqfzgolr"
+      # FIXME: add to GitHub environments or use S3_REGION
       aws_region:
         description: "AWS region"
         required: false
         type: string
         default: "us-east-1"
-      test-suites:
-        description: "Comma-separated list of test suites to run. No spaces in between."
+      # Supported fields for the test matrix:
+      # - test_suite: (required) The test suite to run. The valid test suites are `os`, `hup`, and `cloud`.
+      # - environment: (required) The balenaCloud environment to use for testing, e.g. `bm.balena-dev.com` or `balena-cloud.com.
+      # - worker_type: The worker type to use for testing. The valid worker types are `qemu` and `testbot`. The default worker type is `testbot`.
+      # - worker_fleets: The testbot fleets for finding available Leviathan workers. Not used for QEMU workers. Can accept a list of apps separated by commas, no spaces in between.
+      # - test_org: The organization to use for testing cloud functionality. This default org is `testbot`.
+      test_matrix:
+        description: "JSON Leviathan test matrix to use for testing. If not provided, the default test matrix will be used."
         required: false
         type: string
-        default: "os,hup,cloud"
-      test-workers:
-        description: "Comma-separated list of worker types to use for testing. No spaces in between. The valid worker types are `qemu` and `testbot`."
-        required: false
-        type: string
-        default: "testbot" # Most devices use "testbot" , so qemu is the exception and should be set at the device repo level
-      worker-fleets:
-        description: "Testbot fleet for finding available Leviathan workers. Not used for QEMU workers. Can accept a list of apps separated by commas, no spaces in between."
-        type: string
-        required: false
-        default: "balena/testbot-rig,balena/testbot-rig-partners,balena/testbot-rig-x86,balena/testbot-rig-partners-x86"
-      # TODO: add BALENA_API_USER as an environment secret and remove the BALENACLOUD_ORG input
-      BALENACLOUD_ORG:
-        description: "User matching the AUTOKIT_BALENACLOUD_API_KEY - requirement for leviathan config.js"
-        type: string
-        required: false
-        default: testbot
+        default: >
+          [
+            {
+              "test_suite": "os",
+              "environment": "bm.balena-dev.com",
+              "worker_type": "testbot"
+            },
+            {
+              "test_suite": "cloud",
+              "environment": "bm.balena-dev.com",
+              "worker_type": "testbot",
+              "test_org": "testbot"
+            },
+            {
+              "test_suite": "hup",
+              "environment": "bm.balena-dev.com",
+              "worker_type": "testbot"
+            },
+          ]
```

Here I think we don't gain anything by setting fields to their defaults.

```suggestion
        default: >
          [
            {
              "test_suite": "os",
              "environment": "bm.balena-dev.com",
            },
            {
              "test_suite": "cloud",
              "environment": "bm.balena-dev.com",
            },
            {
              "test_suite": "hup",
              "environment": "bm.balena-dev.com",
            },
          ]
```

---

agree